src/Admin/SupportLink.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Service Link class that adds support links throughout the plugin's UI.
+ *
+ * @package Ampproject\Ampwp
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use WP_Admin_Bar;
+use WP_Post;
+use AMP_Validated_URL_Post_Type;
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AmpProject\AmpWP\QueryVar;
+
+/**
+ * Service that adds support links throughout the plugin's UI.
+ *
+ * @internal
+ */
+class SupportLink implements Service, Conditional, Registerable {
+
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed() {
+
+		return current_user_can( 'manage_options' );
+	}
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @return void
+	 */
+	public function register() {
+
+		// Add support link to Admin Bar.
+		add_action( 'admin_bar_menu', [ $this, 'admin_bar_menu' ], 105 );
+
+		// Add support link to meta box.
+		add_filter( 'amp_validated_url_status_actions', [ $this, 'amp_validated_url_status_actions' ], 10, 2 );
+
+		// Add support link to Post row actions.
+		add_filter( 'post_row_actions', [ $this, 'post_row_actions' ], PHP_INT_MAX, 2 );
+
+		// Plugin row Support link.
+		add_filter( 'plugin_row_meta', [ $this, 'plugin_row_meta' ], 10, 2 );
+
+	}
+
+	/**
+	 * Add Diagnostic link to Admin Bar.
+	 *
+	 * @param WP_Admin_Bar $wp_admin_bar WP_Admin_Bar object.
+	 *
+	 * @return void
+	 */
+	public function admin_bar_menu( $wp_admin_bar ) {
+
+		if ( empty( $wp_admin_bar ) ||
+			! is_a( $wp_admin_bar, 'WP_Admin_Bar' ) ||
+			! array_key_exists( 'amp', $wp_admin_bar->get_nodes() )
+		) {
+			return;
+		}
+
+		// Get the AMP Validated URL post ID.
+		$current_url = remove_query_arg(
+			array_merge( wp_removable_query_args(), [ QueryVar::NOAMP ] ),
+			amp_get_current_url()
+		);
```

Comment:
The `remove_query_arg()` here is redundant because `get_invalid_url_post()` will call by default `normalize` and thus call `\AMP_Validated_URL_Post_Type::normalize_url_for_storage()` which does the same.

```suggestion
		$current_url = amp_get_current_url();
```